### PR TITLE
add tests and better docs for get_bundle_all_angle

### PR DESCRIPTION
**Pull request by joamatab**
- add tests for get_bundle_all_angle
- more docs for get_bundle_all_angle
- check whether steps have valid instructions

https://github.com/gdsfactory/gdsfactory/issues/1255

@tvt173 
@95-martin-orion

### CODE REVIEW

File: gdsfactory/typings.py [hunk 1]
```diff
@@ -45,6 +45,48 @@
...
+_STEP_ALL_ANGLE = Literal[
+    "x",
+    "y",
+    "dx",
+    "dy",
+    "ds",
+    "exit_angle",
+    "cross_section",
+    "connector",
+    "separation",
```
Comment (95-martin-orion) [inline gdsfactory/typings.py:83]: A key part of #1255 is documenting what these fields actually do. It's fine if this is addressed in a separate PR, just wanted to highlight the issue.

File: gdsfactory/typings.py [hunk 2]
```diff
@@ -45,6 +45,48 @@
...
+]
+
+
+Step = Dict[_STEP, float]
+StepAllAngle = Dict[_STEP_ALL_ANGLE, float]
```
Comment (95-martin-orion) [inline gdsfactory/typings.py:88]: You're welcome to ignore this if it doesn't align with gdsfactory style / requirements / etc., but part of why I requested a dataclass for these is that it makes the fields more discoverable. If `Step` has an attribute `dy`, my IDE will autofill that attribute (and possibly display docs for it, if they exist). Keeping these as dicts, even with a `Literal` to enumerate key values, does not provide that benefit.

**Review by 95-martin-orion — CHANGES_REQUESTED** [review]

File: gdsfactory/typings.py (hunk 1 above)
Comment (joamatab) [reply gdsfactory/typings.py:83]: @tvt173 

Troy, can you help us document this?

File: docs/notebooks/042_non-manhattan-router.ipynb [hunk 3]
```diff
@@ -156,6 +156,113 @@
...
+    "c"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import gdsfactory as gf\n",
```
[the remark this answers is not in the document]
Comment (joamatab) [reply docs/notebooks/042_non-manhattan-router.ipynb:217]: I think Troy was going to add more examples :-)

once we have better examples we can remove this one

@SkandanC  and @thomasdorch  have also worked on routing and may also find this interesting

File: gdsfactory/typings.py (hunk 1 above)
Comment (joamatab) [reply gdsfactory/typings.py:83]: see

https://github.com/gdsfactory/gdsfactory/pull/1257/commits/0a717c4b08807086126dc0468c6a73d290a4fb58

File: gdsfactory/typings.py (hunk 2 above)
Comment (joamatab) [reply gdsfactory/typings.py]: see https://github.com/gdsfactory/gdsfactory/pull/1257/commits/0a717c4b08807086126dc0468c6a73d290a4fb58

File: docs/notebooks/042_non-manhattan-router.ipynb (hunk 3 above)
[the remark this answers is not in the document]
Comment (tvt173) [reply docs/notebooks/042_non-manhattan-router.ipynb:217]: yes, i have a bunch more examples i'm in the process of adding :)

File: gdsfactory/typings.py (hunk 2 above)
Comment (tvt173) [reply gdsfactory/typings.py]: as long as it doesn't break the yaml flow, i'm ok

**Pull request merged**